packages/api/src/users/users.controller.ts

```diff
@@ -39,6 +44,16 @@ export class UsersController {
         throw new YoutubeAuthorizationError(ExitCodes.CHANNEL_NOT_FOUND, `No Youtube Channel exists for given user`)
       }
 
+      const [registeredChannel] = await this.channelsService.getAll(user.id)
+
+      // Ensure selected YT channel is not already registered for YPP program
+      if (registeredChannel) {
+        throw new YoutubeAuthorizationError(
+          ExitCodes.CHANNEL_ALREADY_REGISTERED,
+          `Selected Youtube channel is already registered for YPP program`
+        )
```

Comment:
Could we pass `registeredChannel.joystreamChannelId` as the third argument here? We need this information for the authorization failed dialog. 
![image](https://user-images.githubusercontent.com/51168865/203016934-6acc0bf7-407f-467e-bd60-0c84efd58d65.png)

Reply:
Addressed in [99b36d1](https://github.com/Joystream/youtube-synch/pull/57/commits/99b36d159be2977df616229c4b0f42c3c86d2b11)